### ISSUE TITLE
Multiple inheritance with abstract property and NamedTuple

### ISSUE THREAD

**Issue by smackesey** [microsoft/pyright#3597]
`NamedTuple` cannot be used to satisfy abstract property requirements. Not sure if this is a bug or as intended:

```
from abc import ABC
from typing import NamedTuple

class Foo(ABC):

    @property
    @abstractmethod
    def myproperty(self) -> str:
        ...

class MyPropertyVanillaMixin:

    def __init__(self, myproperty: str):
        self._myproperty = myproperty

    @property
    def myproperty(self) -> str:
        return self._myproperty

MyPropertyNamedTupleMixin = NamedTuple("MyPropertyNamedTuple", [("myproperty", str)])

class BarVanilla(MyPropertyVanillaMixin, Foo):
    pass

class BarNamedTuple(MyPropertyNamedTupleMixin, Foo):
    pass

# OK
BarVanilla(myproperty="myproperty")

# ERROR:
# Pyright: Cannot instantiate abstract class BarNamedTuple.
#   "Foo.myproperty" is abstract
BarNamedTuple(myproperty="myproperty")
```

Pyright v1.1.255

**Comment by smackesey** [comment]
I see, thanks for the quick response. As the bug reporter, I am of course biased but I think it would be a shame to ignore this-- mixing `NamedTuple` and abstract classes is a fairly common pattern in our ([Dagster](https://github.com/dagster-io/dagster)) codebase.

**Comment by erictraut** [comment]
OK, I've added some special-case logic to handle this case. This will be included in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.256, which I just published. It will also be included in the next release of pylance.
